src/store/modules/util/actions.ts

```diff
@@ -175,6 +177,55 @@ const actions: ActionTree<UtilState, RootState> = {
 
     commit(types.UTIL_EXTERNAL_MAPPING_TYPES_UPDATED, externalMappingTypes)
   },
+  
+  async fetchCalendars({ commit }, payload) {
+    let calendars = [] as any
+    let calendarWeekTimings = [] as any
+    let resp;
+    
+    try {
+      resp = await UtilService.fetchCalendars({
+        inputFields: {
+          facilityId: payload.facilityId
+        },
+        entityName: "TechDataCalendar",
+        fieldList: ['calendarId', 'calendarWeekId', 'description'],
+        viewSize: 100,
+        noConditionFind: 'Y'
+      })
+      
+      if(!hasError(resp) && resp.data.count) {
+        calendars = resp.data.docs
+
+        resp = await UtilService.fetchCalendarWeekTimings({
+          inputFields: {
+            calendarWeekId: payload.calendarWeekId
```

Comment:
Here we need to fetch calendar weeks based on the calenders fetched in the first step.